cirq-core/cirq/protocols/json_test_data/TrialResult.json

```diff
@@ -0,0 +1,297 @@
+            "name": "a"
+          },
+          0.5
+        ]
+      ]
+    },
+    "measurements": {
+      "m": {
+        "packed_digits": "934e554d5059010076007b276465736372273a20277c6931272c2027666f727472616e5f6f72646572273a2046616c73652c20277368617065273a2028332c2033292c207d202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020200a000102030405060708",
+        "binary": false,
+        "dtype": "int8",
+        "shape": [
+          3,
+          3
+        ]
+      }
+    }
+  },
+  {
+    "cirq_type": "Result",
+    "params": {
+      "cirq_type": "ParamResolver",
+      "param_dict": [
+        [
+          {
+            "cirq_type": "sympy.Symbol",
+            "name": "a"
+          },
+          0.5
+        ]
+      ]
+    },
+    "measurements": {
+      "m": {
+        "packed_digits": "934e554d5059010076007b276465736372273a20273c6932272c2027666f727472616e5f6f72646572273a2046616c73652c20277368617065273a2028332c2033292c207d202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020200a000001000200030004000500060007000800",
+        "binary": false,
+        "dtype": "int16",
+        "shape": [
+          3,
+          3
+        ]
+      }
+    }
+  },
+  {
+    "cirq_type": "Result",
+    "params": {
+      "cirq_type": "ParamResolver",
+      "param_dict": [
+        [
+          {
+            "cirq_type": "sympy.Symbol",
+            "name": "a"
+          },
+          0.5
+        ]
+      ]
+    },
+    "measurements": {
+      "m": {
+        "packed_digits": "934e554d5059010076007b276465736372273a20273c6934272c2027666f727472616e5f6f72646572273a2046616c73652c20277368617065273a2028332c2033292c207d202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020200a000000000100000002000000030000000400000005000000060000000700000008000000",
+        "binary": false,
+        "dtype": "int32",
+        "shape": [
+          3,
+          3
+        ]
+      }
+    }
+  },
+  {
+    "cirq_type": "Result",
+    "params": {
+      "cirq_type": "ParamResolver",
+      "param_dict": [
+        [
+          {
+            "cirq_type": "sympy.Symbol",
+            "name": "a"
+          },
+          0.5
+        ]
+      ]
+    },
+    "measurements": {
+      "m": {
+        "packed_digits": "934e554d5059010076007b276465736372273a20273c6938272c2027666f727472616e5f6f72646572273a2046616c73652c20277368617065273a2028332c2033292c207d202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020200a000000000000000001000000000000000200000000000000030000000000000004000000000000000500000000000000060000000000000007000000000000000800000000000000",
+        "binary": false,
+        "dtype": "int64",
+        "shape": [
+          3,
+          3
+        ]
+      }
+    }
+  },
+  {
+    "cirq_type": "Result",
```

Comment:
this doesn't look like it was saved with an old version of cirq

Reply:
No this file wasn't. The test seemed to do more explicit checking than doing just something like `load_json` and seeing that it worked. It would also look for the `TrialResult` class somewhere which we got rid of and don't want to put back (I don't think) I did check the functionality hit here by doing the following thing:

In Cirq==0.8.0:
```python3
circuit = cirq.Circuit(cirq.H(q), cirq.measure(q))
results = cirq.Simulator().run(circuit)
cirq.to_json(results)
```
Then in cirq master:
```python3
cirq.load_json(json_text=<text from previous snippet>)
```
Which would fail. From there I just added this entry into the resolver cache and it started working again.

Reply:
I think that's what the `_inward` path is for as @balopat suggested

Reply:
Ahhh ok! Went ahead and renamed to _inward and things are looking good now